File: bnctl-banking-android/app/src/main/java/tl/bnctl/banking/data/cards/model/Card.kt
package tl.bnctl.banking.data.cards.model

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class Card(
    var accountId: String?,
    var cardNumber: String?,
    var cardBIN: String?,
    var cardIIN: String?,
    var cardType: String?, // debit or credit
    var cardProduct: String?,
    var cardProductCode: String?,
    var cardAccountNumber: String?, // the account number, associated with this card
    var cardPrintName: String?,
    var cardStatus: String?,
    var expiryDate: String?,
    var currency: String?,
    var cardProductLabel: String?,
    var accountBalance: Double,
    var availableBalance: Double,
    val blockedAmount: Double,
    val approvedOverdraft: Double

) : Parcelable

File: banking-ios/Shared/Model/Cards/Card.swift
//
//  Card.swift
//  laboratory0 (iOS)
//
//  Created by Evgeniy Raev on 7.10.21.
//

import Foundation

struct Card: Codable, Hashable {

    let accountId: String
    let cardNumber: String
    let cardBIN: String
    let cardIIN: String
    let cardType: String
    let cardProduct: String
    let cardProductCode: String
    let cardProductLabel: String
    let cardAccountNumber: String
    let cardPrintName: String
    let cardOwner:String
    let cardStatus: String
    let expiryDate: String
    let currency: String
    let accountBalance: Double
    let availableBalance: Double
    let approvedOverdraft: Double
    let blockedAmount: Double
    let cardId: String
    let cardSecret: String
}
